### PR TITLE
more fixes from cppcheck

### PR DESCRIPTION
**Pull request by cujomalainey**
see commits

### CODE REVIEW

File: src/ipc/handler-ipc3.c [hunk 1]
```diff
@@ -390,7 +390,7 @@ static int ipc_stream_trigger(uint32_t header)
 	struct ipc *ipc = ipc_get();
 	struct ipc_comp_dev *pcm_dev;
 	struct sof_ipc_stream stream;
-	uint32_t ipc_cmd = iCS(header);
+	uint32_t ipc_cmd_hdr = iCS(header);
```
Comment (keyonjie) [inline src/ipc/handler-ipc3.c:393]: ipc_command? it is actually command not header.

File: src/lib/alloc.c [hunk 2]
```diff
@@ -109,7 +109,6 @@ static void write_pattern(struct mm_heap *heap_map, int heap_depth,
 			memset(
 				(void *)current_map->base, pattern,
 				current_map->count * current_map->block_size);
-					       sizeof(*current_map));
```
Comment (keyonjie) [inline src/lib/alloc.c:112]: wow, we must have other not compiled code hidden by conditional compilation.

File: src/ipc/handler-ipc3.c [hunk 3]
```diff
@@ -739,6 +738,7 @@ static int ipc_dma_trace_config(uint32_t header)
 
 error:
 	return err;
+#endif /* CONFIG_SUECREEK || defined __ZEPHYR__ */
```
Comment (keyonjie) [inline src/ipc/handler-ipc3.c:741]: how about to change it like this to make it more readable:
```
#if CONFIG_SUECREEK || defined __ZEPHYR__
static int ipc_dma_trace_config(uint32_t header)
{
	return 0;
}
#else
static int ipc_dma_trace_config(uint32_t header)
{
...
}
#endif
```

**Review by keyonjie — COMMENTED** [review]
Thanks for the cleanup, only minor suggestions, remains looks good to me.

File: src/lib/alloc.c (hunk 2 above)
Comment (cujomalainey) [reply src/lib/alloc.c:112]: one more reason to try and make allyes and rand config builds working

File: src/ipc/handler-ipc3.c (hunk 3 above)
Comment (cujomalainey) [reply src/ipc/handler-ipc3.c:741]: sure works for me

**Review by iuliana-prodan — APPROVED** [review]

File: src/lib/alloc.c (hunk 2 above)
Comment (lgirdwood) [reply src/lib/alloc.c:112]: This is a result of running ```sed``` to remove the platform_commit() call.

**Review by lgirdwood — APPROVED** [review]

**Review by keyonjie — APPROVED** [review]
looks good, thanks.

**Pull request merged**